content/client-server-api/modules/end_to_end_encryption.md

```diff
@@ -1580,10 +1582,36 @@ ratchet index that they have already decrypted. Care should be taken in
 order to avoid false positives, as a client may decrypt the same event
 twice as part of its normal processing.
 
-As with Olm events, clients must confirm that the `sender_key` belongs
-to the user who sent the message. The same reasoning applies, but the
-sender ed25519 key has to be inferred from the `keys.ed25519` property
-of the event which established the Megolm session.
+Similar to Olm events, clients should confirm that the user who sent the
+message corresponds to the user the message was expected to come from.
+For room events, this means ensuring the event's `sender`, `room_id`, and
+the recorded `session_id` match a trusted session (eg: the `session_id`
+is already known and validated to the client).
+
+{{% boxes/note %}}
+As of `v1.3`, the `sender_key` and `device_id` keys are **deprecated**. They
+SHOULD continue to be sent, however they MUST NOT be used to verify the
+message's source.
+
+Clients MUST NOT store or lookup sessions using the `sender_key` or `device_id`.
+
+In a future version of the specification the keys can be removed completely,
+including for sending new messages.
+{{% /boxes/note %}}
+
+{{% boxes/rationale %}}
+Removing the fields (eventually) improves privacy and security by masking the
+device which sent the encrypted message as well as reducing the client's
+dependence on untrusted data: a malicious server (or similar attacker) could
+change these values, and other devices/users can simply lie about them too.
+
+We can remove the fields, particularly the `sender_key`, because the `session_id`
+is already globally unique, therefore making storage and lookup possible without
+the need for added context from the `sender_key` or `device_id`.
```

Comment:
```suggestion
We can remove the fields, particularly the `sender_key` as the `session_id`
is already globally unique. Storage and lookup is therefore possible without
the need for added context from the `sender_key` or `device_id`.
```

Sentence refactoring.

Reply:
err, I think the proposed sentence is a bit inaccurate: the `sender_key` is being removed, not kept, so it's inaccurate to say it's "as" something. The `session_id` is something we already have.

Reply:
I think it was perhaps meant as "as" in the sense of "because" or "since", but removing the comma made it ambiguous.

Reply:
if there's not anything unclear about the current text then I'd prefer to keep it, for "single voice" reasons.